core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1118,6 +1126,23 @@ public boolean isForNewerHudson() {
             }
         }
 
+        /**
+         * Returns true iff the plugin declares a minimum Java version and it's newer than what the Jenkins master is running on.
+         *
+         * @return true iff the plugin declares a minimum Java version and it's newer than what the Jenkins master is running on.
+         *
+         * @since TODO
+         */
+        public boolean isForNewerJava() {
+            try {
+                return minimumJavaVersion != null && new VersionNumber(minimumJavaVersion).isNewerThan(
+                        new VersionNumber(System.getProperty("java.specification.version")));
```

Comment:
between this and [#763](https://github.com/jenkinsci/jenkins/pull/3016/files#diff-dd00e0527600a2b32c10fe2d5af188adR763), could we externalize (in a method) how we get the Java version? This way, we make sure that any part of the code is using the same system and when we will be able to change it by `Runtime.version()` we will have to do it in one place?

Reply:
Yes, absolutely. I almost did it, but didn't want to pull myself into an out of scope change in the same PR. (_looks_ tiny, granted, but devil's in the details...). I had a look, but given Runtime.version() returns a new `Runtime.Version` class, it would not be _such_ a straightforward and forward compatible function.

I would propose to file a separate PR to group the retrievals to java.specification.version (or wrong java.version) as there are many in the core.
Likely, this could even be centralized in the versionnumber-lib.

Reply:
At least a method in `Util` for now.

Reply:
done